python/ray/data/tests/test_execution_optimizer_limit_pushdown.py

```diff
@@ -19,8 +19,21 @@ def _check_valid_plan_and_result(
     expected_plan: Plan,
     expected_result: List[Dict[str, Any]],
     expected_physical_plan_ops=None,
+    check_ordering=True,
 ):
-    assert ds.take_all() == expected_result
+    actual_result = ds.take_all()
+    if check_ordering:
+        assert actual_result == expected_result
+    else:
+        # When ordering doesn't matter, compare as multisets.
+        from collections import Counter
+
+        def to_hashable(d):
+            return tuple(sorted(d.items()))
+
+        assert Counter(map(to_hashable, actual_result)) == Counter(
+            map(to_hashable, expected_result)
+        )
```

Comment:
```suggestion
        assert rows_same(pd.DataFrame(actual_result), pd.DataFrame(expected_result))
```

Let's just use `rows_same` from `ray/data/_internal/util.py`

Reply:
Good catch!